File: templates/lang/android_app_dagger/src_files/app/RootBindingModule.java
package {{ .Group }}.app;

import {{ .Group }}.main.MainActivity;
import {{ .Group }}.main.MainActivityModule;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
abstract class RootBindingModule {

  @ContributesAndroidInjector(modules = {MainActivityModule.class})
  abstract MainActivity mainActivity();

}

File: templates/lang/kotlin_android_app_dagger/src_files/app/RootBindingModule.kt
package {{ .Group }}.app


import com.episode6.hackit.android.inject.context.scope.ActivityScope
import com.episode6.hackit.android.inject.context.scope.ContextScope
import {{ .Group }}.main.MainActivity
import {{ .Group }}.main.MainActivityModule
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class RootBindingModule {

  @ActivityScope
  @ContextScope
  @ContributesAndroidInjector(modules = arrayOf(MainActivityModule::class))
  internal abstract fun mainActivity(): MainActivity

}
